src/test/java/org/rulez/demokracia/pdengine/VoteShowValidationTest.java

```diff
@@ -0,0 +1,169 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.assertEquals;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.annotations.tested_behaviour;
+import org.rulez.demokracia.pdengine.annotations.tested_feature;
+import org.rulez.demokracia.pdengine.annotations.tested_operation;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultVoteRegistry;
+
+public class VoteShowValidationTest extends CreatedDefaultVoteRegistry {
+
+	Vote vote;
+	String voteId;
+	String adminKey;
+	
+	public void setUp() throws ReportedException {
+		super.setUp();
+		vote = voteManager.getVote(adminInfo.voteId);
+		voteId = adminInfo.voteId;
+		adminKey = adminInfo.adminKey;
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_voteId_is_rejected() throws ReportedException {
+		String invalidvoteId = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(invalidvoteId, adminInfo.adminKey)
+		).assertMessageIs(String.format("illegal voteId: %s",invalidvoteId));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("validates inputs")
+	@Test
+	public void invalid_adminKey_is_rejected() throws ReportedException {
+		String invalidAdminKey = RandomUtils.createRandomKey();
+		assertThrows(
+			() -> voteManager.showVote(adminInfo.voteId, invalidAdminKey)
+		).assertMessageIs(String.format("Illegal adminKey: %s",invalidAdminKey));
+	}
+	
+	@tested_feature("Manage votes")
+	@tested_operation("show vote")
+	@tested_behaviour("returns the vote in json")
+	@Test
+	public void check_name() throws ReportedException {
```

Comment:
this is still not a proper sentence, like "the name attribute contains the name of the vote"

Reply:
Fixed